test/integration/022_bigquery_test/test_bigquery_adapter_specific.py

```diff
@@ -0,0 +1,37 @@
+""""Test adapter specific config options."""
+from test.integration.base import DBTIntegrationTest, use_profile
+import textwrap
+import yaml
+
+
+class TestBigqueryDatePartitioning(DBTIntegrationTest):
+
+    @property
+    def schema(self):
+        return "bigquery_test_022"
+
+    @property
+    def models(self):
+        return "adapter-specific-models"
+
+    @property
+    def profile_config(self):
+        return self.bigquery_profile()
+
+    @property
+    def project_config(self):
+        return yaml.safe_load(textwrap.dedent('''\
+        config-version: 2
+        models:
+            test:
+                materialized: table
+                expiring_table:
+                    time_to_expiration: 4    
+        '''))
+
+    @use_profile('bigquery')
+    def test_bigquery_location_invalid(self):
+        _, stdout = self.run_dbt_and_capture()
+        self.assertIn(
+            'expiration_timestamp: TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL '
+            '4 hour)', stdout)
```

Comment:
If this is working the way I'm hoping it is (asserts the ddl contains the option and succeeds), I'll probably add a couple more of these tests for other adapter specific configs like `kms_key_name`, etc.

Reply:
If you call it like `self.run_dbt_and_capture(['run', '--debug'])`, I think you'll also be able to get this information. I don't think it'll be there otherwise.